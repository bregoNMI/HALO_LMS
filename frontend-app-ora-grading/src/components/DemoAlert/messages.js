import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  warningMessage: {
    id: 'ora-grading.demoAlert.warningMessage',
    defaultMessage: 'Grade submission is disabled in the Demo mode of the new ORA Staff Grader.',
    description: 'Submit Grade button text after successful submission',
  },
  confirm: {
    id: 'ora-grading.demoAlert.confirm',
    defaultMessage: 'Confirm',
    description: 'Confirm button text',
  },
  title: {
    id: 'ora-grading.demoAlert.title',
    defaultMessage: 'Demo submit prevented',
    description: 'Title of alert modal after submit was prevented because in demo mode',
  },
});

export default messages;
