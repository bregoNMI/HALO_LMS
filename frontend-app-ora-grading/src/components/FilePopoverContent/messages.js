import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  filePopoverNameTitle: {
    id: 'ora-grading.FilePopoverContent.filePopoverNameTitle',
    defaultMessage: 'File Name',
    description: 'Popover title for file name',
  },
  filePopoverDescriptionTitle: {
    id: 'ora-grading.FilePopoverCellContent.filePopoverDescriptionTitle',
    defaultMessage: 'File Description',
    description: 'Popover title for file description',
  },
  fileSizeTitle: {
    id: 'ora-grading.FilePopoverCellContent.fileSizeTitle',
    defaultMessage: 'File Size',
    description: 'Popover title for file size',
  },
});

export default messages;
