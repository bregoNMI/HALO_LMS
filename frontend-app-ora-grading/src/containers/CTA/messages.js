/* eslint-disable quotes */
import { defineMessages } from '@edx/frontend-platform/i18n';
import { StrictDict } from 'utils';

const messages = defineMessages({
  ctaFeedbackMessage: {
    id: 'ora-grading.CTA.feedbackMessage',
    defaultMessage: 'Thanks for using the new ORA staff grading experience. ',
    description: 'Thank user for using ora and ask for feed back',
  },
  ctaLinkMessage: {
    id: 'ora-grading.CTA.linkMessage',
    defaultMessage: 'Provide some feedback',
    description: 'placeholder for the feedback anchor link',
  },
  ctaLetUsKnowMessage: {
    id: 'ora-grading.CTA.letUsKnowMessage',
    defaultMessage: ' and let us know what you think!',
    description: 'inform user to provide feedback',
  },
});

export default StrictDict(messages);
