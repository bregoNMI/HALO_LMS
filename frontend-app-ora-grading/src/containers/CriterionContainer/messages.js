import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  addComments: {
    id: 'ora-grading.CriterionFeedback.addCommentsLabel',
    defaultMessage: 'Add comments',
    description: 'label for editable feedback field',
  },
  comments: {
    id: 'ora-grading.CriterionFeedback.commentsLabel',
    defaultMessage: 'Comments',
    description: 'label for read-only feedback field',
  },
  optional: {
    id: 'ora-grading.CriterionFeedback.optional',
    defaultMessage: '(Optional)',
    description: 'additional label for optional feedback field',
  },
  optionPoints: {
    id: 'ora-grading.RadioCriterion.optionPoints',
    defaultMessage: '{points} points',
    description: 'criterion option point value display',
  },
  rubricSelectedError: {
    id: 'ora-grading.RadioCriterion.rubricSelectedError',
    defaultMessage: 'Rubric selection is required',
    description: 'Error message when rubric radio did not get selected',
  },
  criterionFeedbackError: {
    id: 'ora-grading.CriterionFeedback.criterionFeedbackError',
    defaultMessage: 'The feedback is required',
    description: 'Error message when feedback is required',
  },
});

export default messages;
