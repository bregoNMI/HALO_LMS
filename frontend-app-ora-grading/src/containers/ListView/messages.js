import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  backToResponses: {
    id: 'ora-grading.ListView.ListViewBreadcrumbs.backToResponses',
    defaultMessage: 'Back to all open responses',
    description: 'Breadcrumbs link text to return to ORA list in LMS',
  },
  noResultsFoundTitle: {
    id: 'ora-grading.ListView.noResultsFoundTitle',
    defaultMessage: 'Nothing here yet',
    description: 'Empty table for the submission table title',
  },
  noResultsFoundBody: {
    id: 'ora-grading.ListView.noResultsFoundBody',
    defaultMessage: 'When learners submit responses, they will appear here',
    description: 'Empty table messages',
  },
  viewAllResponses: {
    id: 'ora-grading.ListView.viewAllResponses',
    defaultMessage: 'View all responses',
    description: 'Button text to load all responses for review/grading',
  },
  viewSelectedResponses: {
    id: 'ora-grading.ListView.viewSelectedResponses',
    defaultMessage: 'View selected responses ({value})',
    description: 'Button text to load selected responses for review/grading',
  },
  username: {
    id: 'ora-grading.ListView.tableHeaders.username',
    defaultMessage: 'Username',
    description: 'Username table column header for submission list view',
  },
  teamName: {
    id: 'ora-grading.ListView.tableHeaders.teamName',
    defaultMessage: 'Team name',
    description: 'Team name table column header for submission list view',
  },
  learnerSubmissionDate: {
    id: 'ora-grading.ListView.tableHeaders.learnerSubmissionDate',
    defaultMessage: 'Learner submission date',
    description: 'Learner submission date table column header for submission list view',
  },
  teamSubmissionDate: {
    id: 'ora-grading.ListView.tableHeaders.teamSubmissionDate',
    defaultMessage: 'Team submission date',
    description: 'Team submission date table column header for submission list view',
  },
  grade: {
    id: 'ora-grading.ListView.tableHeaders.grade',
    defaultMessage: 'Grade',
    description: 'Grade table column header for submission list view',
  },
  gradingStatus: {
    id: 'ora-grading.ListView.tableHeaders.gradingStatus',
    defaultMessage: 'Grading status',
    description: 'Grading status table column header for submission list view',
  },
  loadErrorHeading: {
    id: 'ora-grading.ListView.loadErrorHeading',
    defaultMessage: 'Error loading submissions',
    description: 'Initialization failure alert header',
  },
  loadErrorMessage: {
    id: 'ora-grading.ListView.loadErrorMessage1',
    defaultMessage: 'An error occurred while loading the submissions for this response. Try reloading the page or going {backToResponses}.',
    description: 'Initialization failure alert message line 2',
  },
  backToResponsesLowercase: {
    id: 'ora-grading.ListView.backToResponsesLowercase',
    defaultMessage: 'back to all Open Responses',
    description: 'lowercase string for link to list of all open responses in lms',
  },
  reloadSubmissions: {
    id: 'ora-grading.ListView.reloadSubmissions',
    defaultMessage: 'Reload submissions',
    description: 'Reload button text in case of network failure',
  },
  loadingResponses: {
    id: 'ora-grading.ListView.loadingResponses',
    defaultMessage: 'Loading responses',
    description: 'loading text for submission response list',
  },
});

export default messages;
