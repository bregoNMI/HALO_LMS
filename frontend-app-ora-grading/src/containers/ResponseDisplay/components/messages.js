import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  filePopoverNameTitle: {
    id: 'ora-grading.ResponseDisplay.FilePopoverCell.filePopoverNameTitle',
    defaultMessage: 'File Name',
    description: 'Popover title for file name',
  },
  filePopoverDescriptionTitle: {
    id: 'ora-grading.ResponseDisplay.FilePopoverCell.filePopoverDescriptionTitle',
    defaultMessage: 'File Description',
    description: 'Popover title for file description',
  },
});

export default messages;
