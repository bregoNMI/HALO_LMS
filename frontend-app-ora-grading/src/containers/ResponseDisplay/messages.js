import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  tableNameHeader: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.tableNameHeader',
    defaultMessage: 'Name',
    description: 'Table header file name',
  },
  tableExtensionHeader: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.tableExtensionHeader',
    defaultMessage: 'File Extension',
    description: 'Table header file extension',
  },
  tablePopoverHeader: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.tablePopoverHeader',
    defaultMessage: 'File Metadata',
    description: 'Table header for popover file metadata',
  },
  downloadFiles: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.downloadFiles',
    defaultMessage: 'Download files',
    description: 'Download files inactive state label',
  },
  downloading: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.downloading',
    defaultMessage: 'Downloading',
    description: 'Download files pending state label',
  },
  downloaded: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.downloaded',
    defaultMessage: 'Downloaded!',
    description: 'Download files completed state label',
  },
  retryDownload: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.retryDownload',
    defaultMessage: 'Retry download',
    description: 'Download files failed state label',
  },
  submissionFiles: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.submissionFile',
    defaultMessage: 'Submission Files',
    description: 'Total submission files',
  },
  exceedFileSize: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.fileSizeExceed',
    defaultMessage: 'Exceeded the allow download size',
    description: 'Exceed the allow download size error message',
  },
});

export default messages;
