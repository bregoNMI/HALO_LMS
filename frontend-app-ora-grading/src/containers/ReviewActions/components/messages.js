import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  overrideConfirmTitle: {
    id: 'ora-grading.ReviewActions.overrideConfirmTitle',
    defaultMessage: 'Are you sure you want to override this grade?',
    description: 'ORA Grading override confirm modal title',
  },
  overrideConfirmWarning: {
    id: 'ora-grading.ReviewActions.overrideConfirmWarning',
    defaultMessage: 'This cannot be undone.  The learner may have already received their grade.',
    description: 'ORA Grading override confirm modal warning/content text',
  },
  overrideConfirmContinue: {
    id: 'ora-grading.ReviewActions.overrideConfirmContinue',
    defaultMessage: 'Continue grade override',
    description: 'ORA Grading override confirm modal confirm button',
  },
  startGrading: {
    id: 'ora-grading.ReviewActions.StartGradingButton.startGrading',
    defaultMessage: 'Start grading',
    description: 'Review pane button text to start grading',
  },
  overrideGrade: {
    id: 'ora-grading.ReviewActions.StartGradingButton.overrideGrade',
    defaultMessage: 'Override grade',
    description: 'Review pane button text to start grading an already graded submission',
  },
  stopGrading: {
    id: 'ora-grading.ReviewActions.StartGradingButton.stopGrading',
    defaultMessage: 'Stop grading this response',
    description: 'Review pane button text to stop grading',
  },
  confirmStopOverrideTitle: {
    id: 'ora-grading.ReviewActions.StopGradingConfirmModal.override.title',
    defaultMessage: 'Are you sure you want to stop grade override?',
    description: 'ORA stop overriding grade confirm modal title',
  },
  confirmStopGradingTitle: {
    id: 'ora-grading.ReviewActions.StopGradingConfirmModal.title',
    defaultMessage: 'Are you sure you want to stop grading this response?',
    description: 'ORA stop grading confirm modal title',
  },
  confirmStopWarning: {
    id: 'ora-grading.ReviewActions.StopGradingConfirmModal.warning',
    defaultMessage: 'Your progress will be lost.',
    description: 'ORA stop grading confirm modal warning/content text',
  },
  confirmStopOverrideAction: {
    id: 'ora-grading.ReviewActions.StopGradingConfirmModal.override.confirmText',
    defaultMessage: 'Stop grade override',
    description: 'ORA stop overriding grade confirm modal confirm text',
  },
  confirmStopGradingAction: {
    id: 'ora-grading.ReviewActions.StopGradingConfirmModal.confirmText',
    defaultMessage: 'Cancel grading',
    description: 'ORA stop grading confirm modal confirm text',
  },
  goBack: {
    id: 'ora-grading.ReviewActions.goBack',
    defaultMessage: 'Go back',
    description: 'Confirm modal cancel button text',
  },
  loadPrevious: {
    id: 'ora-grading.ReviewActions.loadPrevious',
    defaultMessage: 'Load previous submission',
    description: 'Alt text for submission navigation (to previous submission) button',
  },
  loadNext: {
    id: 'ora-grading.ReviewActions.loadNext',
    defaultMessage: 'Load next submission',
    description: 'Alt text for submission navigation (to next submission) button',
  },
  navigationLabel: {
    id: 'ora-grading.ReviewActions.navigationLabel',
    defaultMessage: '{current} of {total}',
    description: 'Submission navigation location label',
  },
});

export default messages;
