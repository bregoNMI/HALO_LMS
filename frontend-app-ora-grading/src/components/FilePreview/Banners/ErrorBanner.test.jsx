import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import ErrorBanner from './ErrorBanner';

import messages from '../messages';

describe('Error Banner component', () => {
  const children = <p>Abitary Child</p>;

  const props = {
    actions: [
      {
        id: 'action1',
        onClick: jest.fn().mockName('action1.onClick'),
        message: messages.retryButton,
      },
      {
        id: 'action2',
        onClick: jest.fn().mockName('action2.onClick'),
        message: messages.retryButton,
      },
    ],
    headingMessage: messages.unknownError,
    children,
  };

  let el;
  beforeEach(() => {
    el = shallow(<ErrorBanner {...props} />);
  });

  test('snapshot', () => {
    expect(el.snapshot).toMatchSnapshot();
  });

  describe('component', () => {
    test('children node', () => {
      const childElement = el.instance.children[1];
      const child = shallow(children);

      expect(childElement.type).toEqual(child.type);
      expect(childElement.children[0].el).toEqual(child.children[0].el);
    });

    test('verify actions', () => {
      const { actions } = el.instance.findByType('Alert')[0].props;
      expect(actions).toHaveLength(props.actions.length);

      actions.forEach((action, index) => {
        expect(action.type).toEqual('Button');
        expect(action.props.onClick).toEqual(props.actions[index].onClick);
        // action message
        expect(action.props.children.props).toEqual(props.actions[index].message);
      });
    });

    test('verify heading', () => {
      const heading = el.instance.findByType('FormattedMessage')[0];
      expect(heading.props).toEqual(props.headingMessage);
    });
  });
});
