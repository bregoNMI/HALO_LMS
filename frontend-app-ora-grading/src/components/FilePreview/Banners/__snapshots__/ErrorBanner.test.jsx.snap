// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Banner component snapshot 1`] = `
<Alert
  actions={
    Array [
      <Button
        onClick={[MockFunction action1.onClick]}
        variant="outline-primary"
      >
        <FormattedMessage
          defaultMessage="Retry"
          description="Retry button for error in file renderer"
          id="ora-grading.ResponseDisplay.FileRenderer.retryButton"
        />
      </Button>,
      <Button
        onClick={[MockFunction action2.onClick]}
        variant="outline-primary"
      >
        <FormattedMessage
          defaultMessage="Retry"
          description="Retry button for error in file renderer"
          id="ora-grading.ResponseDisplay.FileRenderer.retryButton"
        />
      </Button>,
    ]
  }
  variant="danger"
>
  <Alert.Heading>
    <FormattedMessage
      defaultMessage="Unknown errors"
      description="Unknown errors message"
      id="ora-grading.ResponseDisplay.FileRenderer.unknownError"
    />
  </Alert.Heading>
  <p>
    Abitary Child
  </p>
</Alert>
`;
